src/librbd/DiffIterate.cc

```diff
@@ -264,7 +264,10 @@ int DiffIterate::execute() {
     RWLock::RLocker l(m_image_ctx.snap_lock);
     RWLock::RLocker l2(m_image_ctx.parent_lock);
     uint64_t overlap = end_size;
-    m_image_ctx.get_parent_overlap(from_snap_id, &overlap);
+    if (m_image_ctx.snaps.empty())
+      m_image_ctx.get_parent_overlap(CEPH_NOSNAP, &overlap);
+    else
+      m_image_ctx.get_parent_overlap(from_snap_id, &overlap);
```

Comment:
This still doesn't look correct (and I'm surprised this bug has been hidden for so long). We know ```from_snap_id == 0``` (from the ```if``` statement on line 263. Therefore, it seems like we really want the snap id of the first snapshot (if available) since the overlap can only shrink.

Also, minor nit: braces around the clauses, please.